compiler-rt/lib/sanitizer_common/sanitizer_file.cpp

```diff
@@ -104,6 +105,17 @@ void ReportFile::SetReportPath(const char *path) {
     }
   }
 
+  // This is so we can use the weak definition from sanitizer_getauxval.h
+  if (&getauxval && getauxval(/* AT_SECURE */ 23) != 0 && path &&
```

Comment:
This needs to be under SANITIZER_LINUX, otherwise we don't get the weak definition

Reply:
Hm, no. sorry. You can't depend on libc in this file.

Reply:
I think you need to implement what needs the `getauxval` (e.g. `IsAT_SECURE()`) in `sanitizer_libcdep.cpp` (with a stub for non-linux platforms), and put a stub (that returns `false`) into `sanitizer_common_nolibc.cpp`)